app/dashboard/views/fido_setup.py

```diff
@@ -1,37 +1,43 @@
 import json
 import secrets
 import uuid
+from time import time
 
 import webauthn
 from flask import render_template, flash, redirect, url_for, session
 from flask_login import login_required, current_user
 from flask_wtf import FlaskForm
-from wtforms import HiddenField, validators
+from wtforms import StringField, HiddenField, validators
 
 from app.config import RP_ID, URL
 from app.dashboard.base import dashboard_bp
 from app.extensions import db
 from app.log import LOG
+from app.models import FIDO, RecoveryCode
+from app.dashboard.views.enter_sudo import sudo_required
 
 
 class FidoTokenForm(FlaskForm):
+    key_name = StringField("key_name", validators=[validators.DataRequired()])
     sk_assertion = HiddenField("sk_assertion", validators=[validators.DataRequired()])
 
 
 @dashboard_bp.route("/fido_setup", methods=["GET", "POST"])
 @login_required
+@sudo_required
 def fido_setup():
-    if current_user.fido_enabled():
-        flash("You have already registered your security key", "warning")
-        return redirect(url_for("dashboard.index"))
-
     if not current_user.can_use_fido:
         flash(
             "This feature is currently in invitation-only beta. Please send us an email if you want to try",
             "warning",
         )
         return redirect(url_for("dashboard.index"))
 
+    if current_user.fido_uuid is not None:
+        fido_model = FIDO.filter_by(uuid=current_user.fido_uuid).all()
```

Comment:
Probably a nitpicking here :), I think we can rename `fido_model` to `fidos`

Reply:
https://github.com/simple-login/app/pull/199/commits/7bd97e13b008b4a3b265f44fc96c2cf8907b3b2a